### PR TITLE
Create pull request: add dark, light, and blue themes

### PR DESCRIPTION
**Pull request by donokuda**
Target branch: `don/stub-ui` (Related: https://github.com/github/VisualStudio/pull/241)

This PR adds the dark, light, and blue themes to the extension. Also fixed an issue where the GitHub pane title wasn't updating its color when the theme is switched.

:sparkles: **SCREENSHOTS!!!** :sparkles: 

Blue
vs-blue

Dark
vs-dark

Light
vs-light

Here are also some demos of the hover and click states:

blue-theme dark-theme light-theme

/cc @shana for :eyes: and :thought_balloon:

### CODE REVIEW

File: src/GitHub.VisualStudio/Styles/ThemeDesignTime.xaml [hunk 1]
```diff
@@ -15,9 +15,30 @@
   <SolidColorBrush x:Key="GitHubVsCommandBarSelectedBorder" Color="#FF3399FF" />
   <SolidColorBrush x:Key="GitHubVsSearchBoxBackground" Color="#FF333337" />
   <SolidColorBrush x:Key="GitHubVsWindowText" Color="#FFF1F1F1" />
+  <SolidColorBrush x:Key="GitHubVsTextBoxBorder" Color="#FF3F3F46" />
+  <SolidColorBrush x:Key="GitHubVsTextBoxBackground" Color="#FF2D2D30" />
+
+  <SolidColorBrush x:Key="GitHubVsButtonBackgroundBrush" Color="#FF3F404B" />
```
Comment (shana) [inline src/GitHub.VisualStudio/Styles/ThemeDesignTime.xaml]: You don't need to add colors to `ThemeDesignTime.xaml` if they're already defined in one of the ThemeXX.xaml files. The only things you need to add to the `ThemeDesignTime.xaml` file is colors that are defined in `VsBrush.xaml`, because that file only has dynamic resources that aren't available at design time.

TL;DR: Kill all these brushes except for `GitHubVsTextBoxBorder` and `GitHubVsTextBoxBorder`

File: src/GitHub.VisualStudio/Styles/VsBrush.xaml [hunk 2]
```diff
@@ -10,4 +10,6 @@
   <SolidColorBrush x:Key="GitHubVsCommandBarSelectedBorder" Color="{DynamicResource VsColor.CommandBarSelectedBorder}" />
   <SolidColorBrush x:Key="GitHubVsSearchBoxBackground" Color="{DynamicResource VsColor.SearchBoxBackground}" />
   <SolidColorBrush x:Key="GitHubVsWindowText" Color="{DynamicResource VsColor.WindowText}" />
+  <SolidColorBrush x:Key="GitHubVsTextBoxBorder" Color="{DynamicResource VsColor.BrandedUIBorder}" />
+  <SolidColorBrush x:Key="GitHubVsTextBoxBackground" Color="{DynamicResource VsColor.BrandedUIBackground}" />
```
Comment (shana) [inline src/GitHub.VisualStudio/Styles/VsBrush.xaml]: For brushes from VS, make sure the names match whatever property name you're using in VsColor, so this should be `GitHubVsBrandedUIBorder` instead.

File: src/GitHub.VisualStudio/Styles/ThemeDark.xaml [hunk 3]
```diff
@@ -19,4 +19,26 @@
   <Color x:Key="GitHubPaneTitleColor">#FFFFFFFF</Color>
   <SolidColorBrush x:Key="GitHubPaneTitleBrush" Color="{StaticResource GitHubPaneTitleColor}" />
 
+  <!--
+  <SolidColorBrush x:Key="TabItem.Selected.Background" Color="#000000"/>
+  <SolidColorBrush x:Key="TabItem.Selected.Border" Color="#ACACAC"/>
+  -->
+
+  <SolidColorBrush x:Key="GitHubVsButtonBackgroundBrush" Color="#FF3F404B" />
```
Comment (shana) [inline src/GitHub.VisualStudio/Styles/ThemeDark.xaml]: On second thought, let's reserve the `GitHubVs` prefix for resources that we're getting directly from VisualStudio, and use `GitHub` for resources that we're defining manually, so that it's easy to tell where they're coming from.

File: src/GitHub.VisualStudio/UI/Views/PullRequestCreationView.xaml.cs [hunk 4]
```diff
@@ -17,7 +17,8 @@ public partial class PullRequestCreationView : GenericPullRequestCreationView
         public PullRequestCreationView()
         {
             InitializeComponent();
-            DataContextChanged += (s, e) => ViewModel = e.NewValue as IPullRequestCreationViewModel;
+            // DataContextChanged += (s, e) => ViewModel = e.NewValue as IPullRequestCreationViewModel;
+            DataContextChanged += (s, e) => ViewModel = new GitHub.SampleData.PullRequestCreationViewModelDesigner() as IPullRequestCreationViewModel;
```
Comment (shana) [inline src/GitHub.VisualStudio/UI/Views/PullRequestCreationView.xaml.cs]: You can kill this change, it doesn't really do anything valuable and it'll break things when we hook up real data.

**Comment by shana** [comment]
Sorry for taking so long looking at this! Looking good! 👾

File: src/GitHub.VisualStudio/Styles/ThemeDesignTime.xaml (hunk 1 above)
Comment (donokuda) [inline src/GitHub.VisualStudio/Styles/ThemeDesignTime.xaml]: D'oh! That was much redundant of me. Went ahead and :fire: them in e92073c

File: src/GitHub.VisualStudio/Styles/VsBrush.xaml (hunk 2 above)
Comment (donokuda) [inline src/GitHub.VisualStudio/Styles/VsBrush.xaml]: That's an excellent guideline to follow. Updated in 0d0a3f5

File: src/GitHub.VisualStudio/Styles/ThemeDark.xaml (hunk 3 above)
Comment (donokuda) [inline src/GitHub.VisualStudio/Styles/ThemeDark.xaml]: :sparkles: Great idea! Updated in 722f4d9

File: src/GitHub.VisualStudio/UI/Views/PullRequestCreationView.xaml.cs (hunk 4 above)
Comment (donokuda) [inline src/GitHub.VisualStudio/UI/Views/PullRequestCreationView.xaml.cs]: I reverted this in edb905e, but can you double check to make sure this looks good?

**Comment by donokuda** [comment]
@shana :raised_hands: Thanks for taking a look at this and no need to apologize for the wait! Can I get another look to see if this looks good?

**Comment by shana** [comment]
:rocket:
